include/gmsh.tcc

```diff
@@ -28,29 +29,30 @@ void GMSH::read_vertices(const std::string& filename) {
     getline(infile, line);
 
     //! Read vertex coordinates & id
-    for (int i = 0; i < nvertices; ++i) {
+    for (unsigned i = 0; i < nvertices; ++i) {
       std::getline(infile, line);
       std::istringstream istream(line);
 
       if (line.find('#') == std::string::npos && line != "") {
         //! Coordinates of vertex
-        std::array<double, ndimension> vertex;
-        
+        std::array<double, Tdim> vertex;
+
         istream >> vertid;
         istream >> vertex.at(0) >> vertex.at(1) >> vertex.at(2);
-
-        vertices_.emplace_back(new Point<ndimension>(vertid, vertex));
+        Mesh<Tdim, Tvertices>::vertices_.insert(std::make_pair(vertid, vertex));
```

Comment:
Use `using Mesh<Tdim, Tvertices>::vertices_` in the header file, and you can use `vertices_`.